src/backend/distributed/commands/table.c

```diff
@@ -3666,6 +3666,21 @@ ErrorIfUnsupportedAlterTableStmt(AlterTableStmt *alterTableStatement)
 
 #if PG_VERSION_NUM >= PG_VERSION_15
 			case AT_SetAccessMethod:
+			{
+				/*
+				 * If command->name == NULL, that means the user is trying to use
+				 * ALTER TABLE ... SET ACCESS METHOD DEFAULT
+				 * which we don't support currently.
+				 */
+				if (command->name == NULL)
+				{
+					ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+									errmsg("ALTER TABLE .. SET ACCESS METHOD DEFAULT "
+										   "commands are currently unsupported.")));
```

Comment:
This looks good thanks. We can explicitly explain to the user that it's just the DEFAULT option that doesn't work, and we can provide a hint also.
```suggestion
ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
									errmsg("DEFAULT option in ALTER TABLE ... SET ACCESS METHOD "
										   "is currently unsupported."),
									errhint("You can rerun the command by explicitly writing the access method name.")));
```